docs/architecture/adr-d001-inter-validator-set-messaging.md

```diff
@@ -0,0 +1,237 @@
+
+## Changelog <!-- omit in toc -->
+
+* 2021-09-29: Initial version of the document
+* 2021-10-05: Clarification of the document based on peer review
+
+## Table of contents <!-- omit in toc -->
+ 
+- [ADR D001: Inter validator set messaging](#adr-d001-inter-validator-set-messaging)
+  - [Context](#context)
+    - [Definitions](#definitions)
+    - [Problem statement](#problem-statement)
+  - [Alternative Approaches](#alternative-approaches)
+    - [Scenario 1: Trigger rotation from ABCI app passing addresses of all quorum members](#scenario-1-trigger-rotation-from-abci-app-passing-addresses-of-all-quorum-members)
+    - [Scenario 2: Trigger rotation from ABCI app passing quorum id to rotate](#scenario-2-trigger-rotation-from-abci-app-passing-quorum-id-to-rotate)
+    - [Scenario 3: Rotate on Tenderdash](#scenario-3-rotate-on-tenderdash)
+  - [Decision](#decision)
+  - [Detailed Design](#detailed-design)
+    - [What are the user requirements?](#what-are-the-user-requirements)
+    - [What systems will be affected?](#what-systems-will-be-affected)
+    - [What new data structures are needed, what data structures need changes?](#what-new-data-structures-are-needed-what-data-structures-need-changes)
+    - [What new APIs will be needed, what APIs will change?](#what-new-apis-will-be-needed-what-apis-will-change)
+    - [What are the efficiency considerations (time/space)?](#what-are-the-efficiency-considerations-timespace)
+    - [What are the expected access patterns (load/throughput)?](#what-are-the-expected-access-patterns-loadthroughput)
+    - [Are there any logging, monitoring, or observability needs?](#are-there-any-logging-monitoring-or-observability-needs)
+    - [Are there any security considerations?](#are-there-any-security-considerations)
+    - [Are there any privacy considerations?](#are-there-any-privacy-considerations)
+    - [How will the changes be tested?](#how-will-the-changes-be-tested)
+    - [How will the changes be broken up for ease of review?](#how-will-the-changes-be-broken-up-for-ease-of-review)
+    - [Will these changes require a breaking (major) release?](#will-these-changes-require-a-breaking-major-release)
+    - [Does this change require coordination with the SDK or any other software?](#does-this-change-require-coordination-with-the-sdk-or-any-other-software)
+  - [Status](#status)
+  - [Consequences](#consequences)
+    - [Positive](#positive)
+    - [Negative](#negative)
+    - [Neutral](#neutral)
+  - [References](#references)
+
+## Context
+
+### Definitions
+
+For the needs of this document, we define the following types of nodes:
+
+* Validator nodes, which participate in the consensus protocol, verify and sign blocks, 
+* Full nodes, which verify and execute blocks. Full nodes verify the recovered threshold signature of the block
+commit message against the current validator set's quorum threshold public key.
+
+Note: there are also light nodes, but light nodes are not relevant for the discussion.
```

Comment:
what do you think, should we mention about sentry node? (which is also not relevant)